examples/rcn.py

```diff
@@ -0,0 +1,390 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:percent
+#     text_representation:
+#       extension: .py
+#       format_name: percent
+#       format_version: '1.3'
+#       jupytext_version: 1.11.4
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# %%
+# %matplotlib inline
+import os
+import time
+
+import jax
+import matplotlib.pyplot as plt
+import numpy as np
+from jax import numpy as jnp
+from jax import tree_util
+from scipy.ndimage import maximum_filter
+from scipy.signal import fftconvolve
+from sklearn.datasets import fetch_openml
+
+from pgmax.fg import graph, groups
+
+os.environ["XLA_PYTHON_CLIENT_ALLOCATOR"] = "platform"
+os.environ["XLA_PYTHON_CLIENT_PREALLOCATE"] = "false"
+
+# %% [markdown]
+# # 1. Load the data
+# %%
+hps, vps = 12, 12
+
+# Use train_size = 100 if you have a gpu with atleast 8Gbs memory.
+# Recommend that jax is installed with cuda enabled for this option.
+train_size = 100
+test_size = 20
+
+
+def fetch_mnist_dataset(test_size: int, seed: int = 5):
+    """Returns test images sampled randomly from the set of MNIST images.
+    Args:
+        test_size: Desired number of test images.
+    Returns:
+        test_set: A list of length test_size containing images from the MNIST test dataset.
+        test_labels: Corresponding labels for the test images.
+    """
+
+    mnist_train_size = 60000
+    num_per_class = test_size // 10
+
+    print("Fetching the MNIST dataset")
+    dataset = fetch_openml("mnist_784", as_frame=False, cache=True)
+    print("Successfully downloaded the MNIST dataset")
+
+    mnist_images = dataset["data"]
+    mnist_labels = dataset["target"].astype("int")
+    full_mnist_test_images = mnist_images[mnist_train_size:]
+    full_mnist_test_labels = mnist_labels[mnist_train_size:]
+
+    np.random.seed(seed)
+
+    test_set = []
+    test_labels = []
+    for i in range(10):
+        idxs = np.random.choice(
+            np.argwhere(full_mnist_test_labels == i)[:, 0], num_per_class
+        )
+        for idx in idxs:
+            img = full_mnist_test_images[idx].reshape(28, 28)
+            img_arr = jax.image.resize(image=img, shape=(112, 112), method="bicubic")
+            img = jnp.pad(
+                img_arr,
+                pad_width=tuple([(p, p) for p in (44, 44)]),
+                mode="constant",
+                constant_values=0,
+            )
+
+            test_set.append(img)
+            test_labels.append(i)
+
+    return test_set, np.array(test_labels)
+
+
+# %%
+test_set, test_labels = fetch_mnist_dataset(test_size)
+train_labels = (
```

Comment:
I tried subsetting a set of trained models, and evaluation breaks completely. In general this seems a very unreliable way to get training labels. Please update `example_data/rcn_100.npz` to include:
1. The 100 training images
2. The labels for the 100 training images
3. The 100 trained RCN models

And load the `train_labels` from `rcn_100.npz`.

Reply:
Updated. The file is now called `rcn.npz` and train sizes below 100 can be sub-sampled.